File: SalonApp/src/main/java/com/salon/service/SalonService.java
package com.salon.service;

import com.salon.model.Salon;
import com.salon.model.Service;

public interface SalonService {
	
	//add salon
	public int createSalon(Salon s);
	
	//get salon
	public Salon getSalonDetails(Long pk);
	
	//deactivate salon
	public String deactivateSalon(Long pk);
	
	//update salon details
	public Salon updateSalonDetails(Long pk, Salon salon);
	
	//add a service to a salon
	public int createService(Service serv, Salon salon);
	
	//deactivate a service for a salon
	//public int deactivateService();
	
	//update a service's details
	
	//get recent orders for salon
	
	//get all reviews

}

File: frontend/SalonApp/src/app/service/salon.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { UserMapping } from './auth-service.service';
import { Customer1 } from './customer-service.service';

//for the salon object
export class Salon {
  constructor(
    public pk:number,
    public salonName:string,
    public address:string, 
    public contact1:string,
    public contact2:string,
    public description:string,
    public salonType :number,
    public homeServiceFlag:number,
    public activeDeactive:number,
    public imagePath:string
  
    ){}
}

export class Service {
  constructor() {}
}

export class Category {
  constructor() {}
}

@Injectable({
  providedIn: 'root'
})
export class SalonService {

  constructor(private httpClient: HttpClient) { }

  public getSalonDetails(pk) {
    console.log("salon rest api call");
    return this.httpClient.get<Salon>('http://localhost:8080/salon/salons/profile/'+pk);
  }

  public deactivateSalon(salon) {
    return this.httpClient.get<string>('http://localhost:8080/salon/salons/deactivate/' + salon.pk);
  }

  //not sending users pk from here.Manage it on server side
  public addSalon(salon) {
    return this.httpClient.post<number>('http://localhost:8080/salon/salons/add',salon);
  }

  public updateSalonDetails(salon) {
    return this.httpClient.put<Salon>('http://localhost:8080/salon/Salons/update',salon);
  }
}
